### ISSUE TITLE
Global variables does not work anymore for file output

### ISSUE THREAD

**Issue by jbfuzier** [balabit/syslog-ng#2927]
Hi,

I just changed from v3.9 to 3.23. 

Global variables in file destination path used to work but not anymore.

## Version of syslog-ng

```
syslog-ng 3 (3.23.1)
Config version: 3.23
Installer-Version: 3.23.1
Revision:
Compile-Date: Sep 11 2019 17:44:21
Module-Directory: /logs/lvapp/lib/syslog-ng
Module-Path: /logs/lvapp/lib/syslog-ng
Include-Path: /logs/lvapp/share/syslog-ng/include
Available-Modules: cef,linux-kmsg-format,confgen,xml,map-value-pairs,kafka,graphite,csvparser,afsocket,pseudofile,afstomp,cryptofuncs,tfgetent,json-plugin,date,appmodel,afuser,system-source,disk-buffer,basicfuncs,afprog,examples,http,hook-commands,syslogformat,snmptrapd-parser,affile,stardate,tags-parser,kvformat,add-contextual-data,dbparser
Enable-Debug: off
Enable-GProf: off
Enable-Memtrace: off
Enable-IPv6: off
Enable-Spoof-Source: off
Enable-TCP-Wrapper: off
Enable-Linux-Caps: off
Enable-Systemd: off
```

## Platform

[root@rhel6jb ~]# uname -a
Linux rhel6jb 2.6.32-504.el6.x86_64 #1 SMP Tue Sep 16 01:56:35 EDT 2014 x86_64 x86_64 x86_64 GNU/Linux
[root@rhel6jb ~]# cat /etc/redhat-release
Red Hat Enterprise Linux Server release 6.6 (Santiago)

# Issue

## Steps to reproduce

1. start syslog-ng

## Configuration

```
@version: 3.23
@include "scl.conf"
source s_local {
        internal();
};

@define a_syslog_log_root_path "foo"
destination d_local {
        file(/tmp/`a_syslog_log_root_path`);
};

log {
        source(s_local);
        destination(d_local);
};
```

## Input and output logs (if possible)
```
lsof :
syslog-ng 2257 root   13w   REG              253,0     1174  1463417 /tmp/`a_syslog_log_root_path`

```

My real config is a little bit more complex, with conf files including other and multiple variable substitutions, but this basic example also shows the issue.

```
# A more usefull example :
## In main syslog-ng.conf :
@define syslog_log_root_path "/tmp/syslog-ng"
@define naming_scheme "${YEAR}_${MONTH}_${DAY}.log"
##In instance1.conf :
@define instance1_syslog_log_root_path "`syslog_log_root_path`/network"
destination d_network_dhcp  {
    file("`instance1_syslog_log_root_path`/dhcp/$FULLHOST/`naming_scheme`"  group("logs_dhcp") dir-group("logs_dhcp") log-fifo-size(500000));
};
```

**Comment by Kokan** [comment]
See https://github.com/balabit/syslog-ng/issues/2906

**Comment by jbfuzier** [comment]
Thanks.
Sorry I missed the open issue.

**Comment by MrAnno** [comment]
Thanks for reporting this issue!

It will be fixed in syslog-ng 3.24.1.

**Comment by alltilla** [comment]
Duplicate of #2906
